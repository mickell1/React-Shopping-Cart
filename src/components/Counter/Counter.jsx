import React, { Component } from "react";
import PropTypes from "prop-types";

import './Counter.scss';

class Counter extends Component {
  constructor(props) {
    super(props);
    this.state = { value: this.props.productQuantity };
    this.increase = this.increase.bind(this);
    this.decrease = this.decrease.bind(this);
  }

  increase(e) {
    this.setState(
      prevState => ({
        value: prevState.value + 1
      }),
      () => {
        this.props.updateQuantity(this.state.value);
      }
    );
    e.preventDefault();
  }

  decrease(e) {
    e.preventDefault();
    if (this.state.value <= 1) {
      return this.state.value;
    } else {
      this.setState(
        prevState => ({
          value: prevState.value - 1
        }),
        () => {
          this.props.updateQuantity(this.state.value);
        }
      );
    }
  }

  feed() {
    this.setState(
      {
        value: this.refs.feedQty.value
      },
      () => {
        this.props.updateQuantity(this.state.value);
      }
    );
  }

  resetQuantity() {
    this.setState({
      value: 1
    });
  }

  render() {
    return (
      <div className="step-input">
        <a href="/#" className="decrease" onClick={this.decrease}>
          â€“
        </a>
        <input
          ref="feedQty"
          type="number"
          className="quantity"
          value={this.state.value}
          onChange={this.feed.bind(this)}
        />
        <a href="/#" className="increase" onClick={this.increase}>
          +
        </a>
      </div>
    );
  }
}

Counter.propTypes = {
  value: PropTypes.number
};

export default Counter;
