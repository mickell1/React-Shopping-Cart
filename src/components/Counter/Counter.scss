@import '../../variables.scss';

.step-input {
  display: flex;
  display: -webkit-flex;
  color: $color-gray-medium;
  max-width: 120px;
  margin: 0 auto;
  .increase,
  .decrease {
    height: 24px;
    width: 24px;
    border: 1px solid $color-gray-dark;
    text-align: center;
    box-sizing: border-box;
    border-radius: 50%;
    text-decoration: none;
    color: inherit;
    font-size: 24px;
    line-height: 22px;
    -moz-user-select: none;
    -webkit-user-select: none;
    &:hover {
      color: $color-red;
      border-color: $color-red;
    }
    &:active {
      color: $color-white;
      border-color: $color-red;
      background: lighten($color-red, 15%);
    }
  }
  .quantity {
    height: 24px;
    width: 48px;
    text-align: center;
    margin: 0 12px;
    border-radius: 2px;
    border: 1px solid $color-gray-dark;
    &:focus {
      outline: none;
      border-color: $color-light-green;
    }
  }
}
