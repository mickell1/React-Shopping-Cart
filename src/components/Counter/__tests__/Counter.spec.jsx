import React from 'react';
import { shallow } from 'enzyme';
import Counter from '../Counter';

describe('Counter', () => {
  let wrapper;

  const defaultProps = {
    state: {
      value: 1,
    },
    updateQuantity: jest.fn(),
  }

  beforeEach(() => {
    wrapper = shallow(<Counter {...defaultProps} /> )  
  })

  it('should render the Counter Component correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });
});