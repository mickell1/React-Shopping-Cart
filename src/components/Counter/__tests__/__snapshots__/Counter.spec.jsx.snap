// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render the Counter Component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter
    state={
      Object {
        "value": 1,
      }
    }
    updateQuantity={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          className="decrease"
          href="/#"
          onClick={[Function]}
        >
          –
        </a>,
        <input
          className="quantity"
          onChange={[Function]}
          type="number"
        />,
        <a
          className="increase"
          href="/#"
          onClick={[Function]}
        >
          +
        </a>,
      ],
      "className": "step-input",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "–",
          "className": "decrease",
          "href": "/#",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "–",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "quantity",
          "onChange": [Function],
          "type": "number",
          "value": undefined,
        },
        "ref": "feedQty",
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+",
          "className": "increase",
          "href": "/#",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "+",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="decrease"
            href="/#"
            onClick={[Function]}
          >
            –
          </a>,
          <input
            className="quantity"
            onChange={[Function]}
            type="number"
          />,
          <a
            className="increase"
            href="/#"
            onClick={[Function]}
          >
            +
          </a>,
        ],
        "className": "step-input",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "–",
            "className": "decrease",
            "href": "/#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "–",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "quantity",
            "onChange": [Function],
            "type": "number",
            "value": undefined,
          },
          "ref": "feedQty",
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+",
            "className": "increase",
            "href": "/#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "+",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
